app/v2/controllers/V2MovementsController.scala

```diff
@@ -467,6 +468,13 @@ class V2MovementsControllerImpl @Inject() (
         EitherT.rightT(())
     }
 
+  def determineMaxPerPageCount(parameter: Option[Long]): ItemCount =
+    parameter match {
+      case Some(x) if x <= config.maxItemsPerPage => ItemCount(x)
+      case Some(x) if x > config.maxItemsPerPage  => ItemCount(config.maxItemsPerPage)
+      case None                                   => ItemCount(config.defaultItemsPerPage)
+    }
```

Comment:
Kind of straying into optimisation territory here, but I don't really like this pattern match because it's doing a type match when it doesn't need to (plus one of the if guards is useless).

I wonder if we'd be better served by calling

```scala
parameter
    .map(count => Math.min(config.maxItemsPerPage, count))
   .getOrElse(ItemCount(config.defaultItemsPerPage))
```

That way we avoid reflection in the match and the runtime can make better decisions about how to make the comparison.